lib/IRGen/GenClass.cpp

```diff
@@ -1908,9 +1909,17 @@ namespace {
       else
         (void)0;
       
-      // If the property is an instance property and has storage, emit the ivar
-      // name last.
-      if (!prop->isStatic() && prop->hasStorage())
+      // If the property is an instance property and has storage, and meanwhile
+      // its type is either trivially representable in ObjC or static bridged,
+      // emit the ivar name last.
+      constexpr auto objcLang = ForeignLanguage::ObjectiveC;
+      bool isTriviallyRepresentable =
+          propTy->isTriviallyRepresentableIn(objcLang, propDC);
+      bool isStaticBridged =
+          ForeignRepresentableKind::StaticBridged ==
+              propTy->getForeignRepresentableIn(objcLang, propDC).first;
```

Comment:
I don't think static bridging counts. It looks like we use that for the closure <-> block bridge, and that's still going to be stored as a closure in memory.

Reply:
Done!